File: bahasaKorea/app/src/main/java/com/im/daeseong/bahasakorea/SplashActivity.java
package com.im.daeseong.bahasakorea;

import android.content.Context;
import android.content.Intent;
import android.content.pm.ActivityInfo;
import android.os.AsyncTask;
import android.os.Handler;
import android.os.Message;
import androidx.appcompat.app.AppCompatActivity;//import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import com.im.daeseong.bahasakorea.Controls.RecycleUtil;
import com.im.daeseong.bahasakorea.Controls.SharedPreferencesUtil;
import com.im.daeseong.bahasakorea.Database.CopyDBfile;

public class SplashActivity extends AppCompatActivity {

    private static final String TAG = SplashActivity.class.getSimpleName();

    private Handler handler = null;

    private View mainView;
    private int nSkinStyle;

    //private KataManager kataManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        InitTitleBar();
        setContentView(R.layout.activity_splash);
        init();

        InitSkinStyle();

        InitData();
    }

    private void init(){

        handler = new Handler() {
            @Override
            public void handleMessage(Message msg) {
                switch (msg.what) {
                    case 0:
                        startActivity(new Intent(SplashActivity.this, MainActivity.class));
                        SplashActivity.this.finish();
                        break;
                }
            }
        };

        mainView = findViewById(android.R.id.content);

    }

    private void InitSkinStyle()
    {
        try {

            nSkinStyle = BahasaApplication.getInstance().getSkinStyle();
            if (nSkinStyle == 1) {
                mainView.setBackgroundResource(R.color.SkinStyle1);//mainView.setBackgroundColor(Color.parseColor("#1B2836"));
            } else if (nSkinStyle == 2) {
                mainView.setBackgroundResource(R.color.SkinStyle2);//mainView.setBackgroundColor(Color.parseColor("#33A7D6"));
            } else if (nSkinStyle == 3) {
                mainView.setBackgroundResource(R.color.SkinStyle3);//mainView.setBackgroundColor(Color.parseColor("#493335"));
            } else if (nSkinStyle == 4) {
                mainView.setBackgroundResource(R.color.SkinStyle4);//mainView.setBackgroundColor(Color.parseColor("#FF80AB"));
            } else if (nSkinStyle == 5) {
                mainView.setBackgroundResource(R.color.SkinStyle5);//mainView.setBackgroundColor(Color.parseColor("#4CAF50"));
            } else if (nSkinStyle == 6) {
                mainView.setBackgroundResource(R.color.SkinStyle6);//mainView.setBackgroundColor(Color.parseColor("#3F51B5"));
            } else if (nSkinStyle == 7) {
                mainView.setBackgroundResource(R.color.SkinStyle7);//mainView.setBackgroundColor(Color.parseColor("#B71C1C"));
            } else if (nSkinStyle == 8) {
                mainView.setBackgroundResource(R.color.SkinStyle8);//mainView.setBackgroundColor(Color.parseColor("#37474F"));
            }

        }catch (Exception e){
        }
    }

    private void InitData(){

        //kataManager = KataManager.getInstance(this);

        try{

            boolean bCopy = new CopyDBTask().execute(SplashActivity.this).get();

        }catch (Exception e){
        }

        handler.postDelayed(new Runnable() {
            @Override
            public void run() {

                try{

                    //스킨 타입
                    int nSkinStyle = (int) SharedPreferencesUtil.getInstance().getValue("SkinStyle", 1);
                    BahasaApplication.getInstance().setSkinStyle(nSkinStyle);

                    //가나다라... 자동 반복 시간
                    int nAlfabetRepeatTime = (int) SharedPreferencesUtil.getInstance().getValue("AlfabetRepeatTime", 10);
                    BahasaApplication.getInstance().setAlfabetRepeatTime(nAlfabetRepeatTime);

                    //음성사용 여부
                    boolean bAlfabetRepeatSound = (boolean) SharedPreferencesUtil.getInstance().getValue("AlfabetRepeatSound", false);
                    BahasaApplication.getInstance().setAlfabetRepeatSound(bAlfabetRepeatSound);

                    //단어장 자동 반복 시간
                    int nIngatRepeatTime = (int) SharedPreferencesUtil.getInstance().getValue("IngatRepeatTime", 10);
                    BahasaApplication.getInstance().setIngatRepeatTime(nIngatRepeatTime);

                    //단어장 음성사용 여부
                    boolean bIngatRepeatSound = (boolean) SharedPreferencesUtil.getInstance().getValue("IngatRepeatSound", false);
                    BahasaApplication.getInstance().setIngatRepeatSound(bIngatRepeatSound);

                    //스크린 화면 락 처리여부
                    boolean bScreenLock = (boolean) SharedPreferencesUtil.getInstance().getValue("ScreenLock", false);
                    BahasaApplication.getInstance().setScreenLock(bScreenLock);


                    //API 이용시만 사용
                    /*
                    //MainTab2Fragment 데이터
                    hangulApi.getInstance().clear();
                    hangulApi.getInstance().InitData();
                    */

                    //API 이용시만 사용
                    /*
                    //MainTab3Fragment 데이터
                    kataApi.getInstance().clear();
                    kataApi.getInstance().InitData(kataManager.getKata());
                    */

                    Message msg = handler.obtainMessage();
                    msg.what = 0;
                    handler.sendMessage(msg);
                    /*
                    runOnUiThread(new Runnable() {
                        @Override
                        public void run() {
                            startActivity(new Intent(SplashActivity.this, MainActivity.class));
                            SplashActivity.this.finish();
                        }
                    });
                    */
                }catch (Exception e){
                }
            }
        }, 1000);
    }

    private void InitTitleBar(){
        setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);

        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
    }

    @Override
    public void onBackPressed() {
        //super.onBackPressed();
        //백버튼 기능 막음
        return;
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();

        try {

            if (handler != null) {
                handler.removeCallbacksAndMessages(null);
                handler = null;
            }

            RecycleUtil.recursiveRecycle(getWindow().getDecorView());
            System.gc();

        }catch (Exception e){
        }
    }

    private static class CopyDBTask extends AsyncTask<Context, Void, Boolean> {

        @Override
        protected Boolean doInBackground(Context... params) {
            try {

                new CopyDBfile(params[0]);
                return true;

            } catch (Exception e) {
            }
            return false;
        }
    }

}

File: bahasakorea(kotlin)/app/src/main/java/com/daeseong/bahasakorea/SplashActivity.kt
package com.daeseong.bahasakorea

import android.content.Context
import android.content.Intent
import android.content.pm.ActivityInfo
import android.os.AsyncTask
import android.os.Bundle
import android.os.Handler
import android.os.Message
import android.view.View
import android.view.Window
import android.view.WindowManager
import androidx.appcompat.app.AppCompatActivity
import com.daeseong.bahasakorea.Controls.RecycleUtil
import com.daeseong.bahasakorea.Controls.SharedPreferencesUtil
import com.daeseong.bahasakorea.Database.CopyDBfile

class SplashActivity : AppCompatActivity() {

    private val tag = SplashActivity::class.java.simpleName

    private var handler: Handler? = null

    private var mainView: View? = null
    private var nSkinStyle = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        InitTitleBar()
        setContentView(R.layout.activity_splash)
        init()

        InitSkinStyle()

        InitData()
    }

    private fun init() {

        mainView = findViewById(android.R.id.content)

        //스킨 타입
        val nSkinStyle = SharedPreferencesUtil.getInstance().getValue("SkinStyle", 1) as Int
        BahasaApplication.getInstance().setSkinStyle(nSkinStyle)

        //가나다라... 자동 반복 시간
        val nAlfabetRepeatTime = SharedPreferencesUtil.getInstance().getValue("AlfabetRepeatTime", 10) as Int
        BahasaApplication.getInstance().setAlfabetRepeatTime(nAlfabetRepeatTime)

        //음성사용 여부
        val bAlfabetRepeatSound = SharedPreferencesUtil.getInstance().getValue("AlfabetRepeatSound", false) as Boolean
        BahasaApplication.getInstance().setAlfabetRepeatSound(bAlfabetRepeatSound)

        //단어장 자동 반복 시간
        val nIngatRepeatTime = SharedPreferencesUtil.getInstance().getValue("IngatRepeatTime", 10) as Int
        BahasaApplication.getInstance().setIngatRepeatTime(nIngatRepeatTime)

        //단어장 음성사용 여부
        val bIngatRepeatSound = SharedPreferencesUtil.getInstance().getValue("IngatRepeatSound", false) as Boolean
        BahasaApplication.getInstance().setIngatRepeatSound(bIngatRepeatSound)

        //스크린 화면 락 처리여부
        val bScreenLock = SharedPreferencesUtil.getInstance().getValue("ScreenLock", false) as Boolean
        BahasaApplication.getInstance().setScreenLock(bScreenLock)

        handler = object : Handler() {
            override fun handleMessage(msg: Message) {
                when (msg.what) {
                    0 -> {
                        startActivity(Intent(this@SplashActivity, MainActivity::class.java))
                        finish()
                    }
                }
            }
        }
    }

    private fun InitSkinStyle() {

        try {
            nSkinStyle = BahasaApplication.getInstance().getSkinStyle()
            when (nSkinStyle) {
                1 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle1)
                }
                2 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle2)
                }
                3 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle3)
                }
                4 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle4)
                }
                5 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle5)
                }
                6 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle6)
                }
                7 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle7)
                }
                8 -> {
                    mainView!!.setBackgroundResource(R.color.SkinStyle8)
                }
            }
        } catch (e: Exception) {
        }
    }

    private fun InitData() {

        try {
            val bCopy = CopyDBTask().execute(this).get()
        } catch (e: java.lang.Exception) {
        }

        handler!!.postDelayed({
            try {

                val msg = handler!!.obtainMessage()
                msg.what = 0
                handler!!.sendMessage(msg)

            } catch (e: java.lang.Exception) {
            }
        }, 1000)
    }

    private fun InitTitleBar() {
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
    }

    override fun onBackPressed() {
        //super.onBackPressed()
        //백버튼 기능 막음
        return
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()

        try {

            if (handler != null) {
                handler!!.removeCallbacksAndMessages(null)
                handler = null
            }

            RecycleUtil.recursiveRecycle(window.decorView)
            System.gc()
        } catch (e: java.lang.Exception) {
        }
    }

    inner class CopyDBTask : AsyncTask<Context?, Void?, Boolean>() {

        override fun doInBackground(vararg params: Context?): Boolean? {

            try {
                CopyDBfile(params[0]!!)
                return true
            } catch (e: java.lang.Exception) {
            }

            return false
        }
    }

}
